cmd/archeio/main_test.go

```diff
@@ -76,30 +88,108 @@ func TestIntegrationMain(t *testing.T) {
 		t.Fatal("timed out waiting for archeio to be ready")
 	}
 
-	// TODO: fake being on AWS
-	testPull := func(image string) {
-		// nolint:gosec // this is not user suplied input ...
-		cmd := exec.Command("crane", "pull", testAddr+"/"+image, os.DevNull)
-		//cmd.Stderr = os.Stderr
-		out, err := cmd.CombinedOutput()
-		if err != nil {
-			t.Errorf("pull for %q failed: %v", image, err)
-			t.Error("output: ")
-			t.Error(string(out))
-			t.Fail()
-		}
+	// perform many test pulls ...
+	testCases := makeTestCases()
+	for i := range testCases {
+		tc := testCases[i]
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			ref := testAddr + "/" + tc.Image
+			// ensure we supply fake IP info from test case
+			craneOpts := []crane.Option{crane.WithTransport(newFakeIPTransport(tc.FakeIP))}
+			// test fetching digest first
+			digest, err := crane.Digest(ref, craneOpts...)
+			if err != nil {
+				t.Errorf("Fetch digest for %q failed: %v", ref, err)
```

Comment:
Nit: Isn't this equivalent to Fatalf?

Reply:
yeah, my copy-paste is showing a bit :-)

Reply:
Actually it's not! `Fatalf` is `Errorf` + `FailNow`, however all of the `Fail` calls are redundant to `Errorf` which is just `Logf` + `Fail`.

Reply:
dropped all the Fail